File: 7_kyu/Alphabet_symmetry.rs
// Count the number of chars with the same position in the string as in the alphabet.
fn count_alphabetics(str: &str) -> usize {
    str.chars()
        .zip('a'..'z')
        .filter(|(s, c)| (*s).to_ascii_lowercase() == *c)
        .count()
}

fn solve(strings: &[String]) -> Vec<usize> {
    strings.iter().map(|str| count_alphabetics(str)).collect()
}

File: 7_kyu/Alphabet_symmetry.lua
alphsymm = {}

-- Count the letter which are at same position as in alphabet.
local function countAlphabetPositions(s)
  s = s:lower()
  local offset = string.byte("a")-1
  local count = 0
  for i = 1,#s do
    if s:sub(i,i):byte() == offset+i then
      count = count + 1
    end
  end
  return count
end

-- Counts the letters which are at same position as in alphabet.
-- Argument
--    Array of strings to count out.
-- Return
--    Array of counts.
function alphsymm.solve(arr)
  local accu = {}
  for i = 1,#arr do
    accu[i] = countAlphabetPositions(arr[i])
  end
  return accu
end

return alphsymm
